### PR TITLE
134 autocomplete

### LINKED ISSUE #173: Fix Angular router (e.g. back button doesn't work sometimes)
To reproduce try downloading a file from a single sequence page after search - going back after that will fail. URL changes, but the page does not.

### PR DESCRIPTION
**Pull request by BurkovBA**
Now module consists of just 2 components: metadataSearchBar and metadataSearchResults, MainController (may be, we could get rid of it too?) and search service (which performs search, displays results and does autocompletion).

Scary and buggy router removed, so... Fixes #173.

### CODE REVIEW

**Comment by AntonPetrov** [comment]
Thank you Boris, this is such a fun feature! A few things I noticed without doing a full review:

* facets like *Organisms* or *RNA type* do not seem to work
* *Load more* button does not seem to work
* perhaps hide angular progress bar for the autocomplete ajax requests?
* maybe show a text-muted message like Google does "Press Enter to search" instead of "No Suggestions"?
* the header jiggles up and down (the height of the blue background changes as one types)
* maybe animate the appearance of the autocomplete?
* it looks like entries with highlighted matches are always shown after matches with no highlights
* *Browse* button does not work on expert database pages like `/expert-database/ensembl`
* some suggestions return no hits (e.g. `rnase11`)

We can work through these things next week. Well done with refactoring so far!

**Comment by BurkovBA** [comment]
Anton, thank you for these notes, very helpful.

Will fix the facets and Browse button on expert database pages first of all. Can't reproduce the "Load more" button failing. Then we can discuss appearance issues.

File: rnacentral/portal/tests/selenium_tests.py [hunk 1]
```diff
@@ -486,82 +543,170 @@ def test_browser_back_button(self):
...
+        # the dict has the following structure:
+        # {expectation: [queries, for which expectation should appear in autocomplete suggestions]}
+        test_suite = OrderedDict([
+            ('mir 12', ['mir-12']),
+            ('lncrna', ['lncrna']),
+            ('mitochondrial', ['mitochondial']),  # sic! - typo is intentional
+            ('kcnq1ot1', ['kcnq1ot1']),
+
+            # key species
+            ('Arabidopsis thaliana', ['Arabidopsis thaliana']),
```
Comment (AntonPetrov) [inline rnacentral/portal/tests/selenium_tests.py]: Not sure I understand why this data structure is needed. The array always has the same value as the first element of the tuple?

File: rnacentral/portal/templates/portal/help/genomic-mapping.html [hunk 2]
```diff
@@ -40,7 +40,7 @@
           <ol>
           {% for species in division.species %}
             <li>
-              <a href="" ng-click="meta_search('TAXONOMY:%22{{ species.taxid }}%22 AND has_genomic_coordinates:%22True%22')">{{ species.name }}</a>
+              <a href="" ng-click="search.search('TAXONOMY:&quot;{{ species.taxid }}&quot; AND has_genomic_coordinates:&quot;True&quot;')">{{ species.name }}</a>
```
Comment (BurkovBA) [inline rnacentral/portal/templates/portal/help/genomic-mapping.html]: "search" is a service, which provides several functions, primarilly "search" and "autocomplete". Well, we can rename it to textSearch, in that case name of the service will match the name of Angular module (when we create it eventually).

Comment (AntonPetrov) [inline rnacentral/portal/templates/portal/help/genomic-mapping.html]: let's rename it, we can find a better name for the Angular module

File: rnacentral/portal/tests/selenium_tests.py (hunk 1 above)
Comment (BurkovBA) [inline rnacentral/portal/tests/selenium_tests.py]: Probably, you're right. I initially wanted 'Arabidopsis thaliana' suggestion to appear both for 'Arabidopsis' and 'Arabidopsis thaliana' queries.

File: rnacentral/portal/templates/portal/help/genomic-mapping.html (hunk 2 above)
Comment (BurkovBA) [inline rnacentral/portal/templates/portal/help/genomic-mapping.html]: I'd just create a separate Angular module, called `textSearch`. I'd put this service and both components into it. I'd leave MainController in RNAcentral module and eventually would've phased it out of the project as much as possible.

File: rnacentral/portal/tests/selenium_tests.py [hunk 3]
```diff
@@ -486,82 +543,170 @@ def test_browser_back_button(self):
...
         page.navigate()
-        self.assertEqual(len(page.get_metasearch_results()), 1)
+        self.assertEqual(len(page.text_search_results), 1)
         # non-existing taxid
-        page = MetaSearchPage(self.browser, 'search?q=URS000047C79B_00000')
+        page = TextSearchPage(self.browser, 'search?q=URS000047C79B_00000')
         page.navigate()
         self.assertTrue(page.warnings_present())
 
+    def test_text_search_test_suite(self):
```
Comment (BurkovBA) [inline rnacentral/portal/tests/selenium_tests.py]: Ok.

**Pull request merged**